rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -651,8 +650,8 @@ class DebruijnInterpreter[M[_], F[_]](
               (keyString -> valueBool.toString).pure[M]
             case (GString(keyString), GUri(uri)) =>
               (keyString -> uri).pure[M]
-            // TODO: Add cases for other ground terms as well? Maybe it would be better
-            // to implement cats.Show for all ground terms.
+            case (GString(keyString), GByteArray(byteString)) =>
+              (keyString -> Base16.encode(byteString.toByteArray)).pure[M]
```

Comment:
I'm a bit partial on having a default format of serializing a byte array to string. Especially that the format is Base16. I'm afraid that this is going to be (ab)used by end users leading to gross inefficiencies. OTOH, it's more rev for the validators, right? :thinking:

Why do we need this?

Reply:
> I’m a bit partial on having a default format of serializing a byte array to string

Honestly, I feel the same way. The only reason it’s Base16 is because that’s Rholang’s default for _serializing_ data in the first place.

> Why do we need this?

It’s a ticket that’s assigned to me and I haven’t issued a PR in a while :) No, in seriousness, @odeac would have a better idea than me. He reported the ticket.

Reply:
Before this the byte array  interpolation was failing silently, which was a bit unexpected. We should support byte array interpolation or, at least, fail with a big fat error saying that we don't support interpolation for this data type.

Reply:
@JosDenmark is working on the failures no longer being silent. Let's close this then.